dask/dataframe/groupby.py

```diff
@@ -1984,6 +1993,45 @@ def rolling(self, window, min_periods=None, center=False, win_type=None, axis=0)
             axis=axis,
         )
 
+    def fillna(self, value=None, method=None, limit=None, axis=None):
+        if not np.isscalar(value) and value is not None:
+            raise NotImplementedError(
+                "groupby-fillna with value=dict/Series/DataFrame is currently not supported"
+            )
+        axis = DataFrame._validate_axis(axis)
+        if value is not None or axis == 0:
```

Comment:
If we use `value` with `fillna` the `axis` doesn't really matter because the output remains the same.

But, that said, I feel if a user specifies `axis=1` they probably assume Dask to be working row-wise internally, so maybe we should use the `apply` implementation whenever `axis=1` irrespective of `value`?

Reply:
I think your choice here is fine, I don't want to belabor the point of `apply` here, it's more of an implementation detail, and shouldn't really affect the user much.